packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -167,18 +167,37 @@ def test_if_overlay_master_is_up(cluster):
 
 
 def test_if_overlay_master_agent_is_up(cluster):
-    r = cluster.get('/mesos/overlay-agent/overlay')
-    assert r.ok, "status_code: {}, content: {}".format(r.status_code, r.content)
+    master_response = cluster.get('/mesos/overlay-master/state')
+    assert master_response.ok,\
+        "status_code: {}, content: {}".format(master_response.status_code, master_response.content)
+
+    master_overlay_json = master_response.json()
+
+    agent_response = cluster.get('/mesos/overlay-agent/overlay')
+    assert agent_response.ok, "status_code: {}, content: {}".format(agent_response.status_code, agent_response.content)
 
     # Make sure the `dcos` overlay has been configured.
-    json = r.json()
+    agent_overlay_json = agent_response.json()
+
+    assert 'ip' in agent_overlay_json
+    agent_ip = agent_overlay_json['ip']
 
-    assert 'overlays' in json
-    assert len(json['overlays']) == 1
+    # Get the Mesos and Docker bridge configuration for this agent from
```

Comment:
@asridharan  - can you move these lines 185 to 192 to line number 212. (It will be much easier to read and follow after this movement).

Everything else looks good to me. So Ship-it from me.